### PR TITLE
Add support for proxies to Recorded Future Connectors

### PR DESCRIPTION
**Pull request by Jonah-RF**
### Proposed changes
For the Recorded Future Notes and Recorded Future Enrichment Connector

* Add support for proxies
* Add support for disabling SSL verification
* Minor improvements to documentation

- [X ] I consider the submitted work as finished
- [ X] I tested the code for its functionality using different use cases
- [ X] I added/update the relevant documentation (either on github or on notion)
- [ X] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: external-import/recordedfuture-notes/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.11-alpine
+FROM python:3.7-alpine
```
Comment (SamuelHassine) [inline external-import/recordedfuture-notes/Dockerfile]: Why?

File: external-import/recordedfuture-notes/docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: '3'
 services:
-  connector-recordedfuture-notes:
-    image: opencti/connector-recordedfuture-notes:5.11.1
+  connector-rf-notes:
+    image: opencti-rf-notes:latest
```
Comment (SamuelHassine) [inline external-import/recordedfuture-notes/docker-compose.yml]: Wrong name

File: external-import/recordedfuture-notes/src/rflib/rf_notes_to_stix2.py [hunk 3]
```diff
@@ -253,23 +235,16 @@ class IntrusionSet(RFStixEntity):
 
     def create_stix_objects(self):
         """Creates STIX objects from object attributes"""
-        self.stix_obj = stix2.IntrusionSet(
-            id=pycti.IntrusionSet.generate_id(self.name),
-            name=self.name,
-            created_by_ref=self.author.id,
-        )
+        self.stix_obj = stix2.v21.IntrusionSet(name=self.name, created_by_ref=self.author.id)
```
Comment (SamuelHassine) [inline external-import/recordedfuture-notes/src/rflib/rf_notes_to_stix2.py]: Need ID generation

File: external-import/recordedfuture-notes/src/rflib/rf_notes_to_stix2.py [hunk 4]
```diff
@@ -253,23 +235,16 @@ class IntrusionSet(RFStixEntity):
...
 
     def create_stix_objects(self):
         """Creates STIX objects from object attributes"""
-        self.stix_obj = stix2.Malware(
-            id=pycti.Malware.generate_id(self.name),
-            name=self.name,
-            is_family=False,
-            created_by_ref=self.author.id,
+        self.stix_obj = stix2.v21.Malware(
+            name=self.name, is_family=False, created_by_ref=self.author.id
```
Comment (SamuelHassine) [inline external-import/recordedfuture-notes/src/rflib/rf_notes_to_stix2.py]: Need ID generation

File: external-import/recordedfuture-notes/src/rflib/rf_notes_to_stix2.py [hunk 5]
```diff
@@ -383,75 +348,66 @@ def __init__(
 
     def _create_author(self):
         """Creates Recorded Future Author"""
-        return stix2.Identity(
-            id=pycti.Identity.generate_id("Recorded Future", "organization"),
-            name="Recorded Future",
-            identity_class="organization",
-        )
+        return stix2.v21.Identity(name='Recorded Future', identity_class='organization')
```
Comment (SamuelHassine) [inline external-import/recordedfuture-notes/src/rflib/rf_notes_to_stix2.py]: Need ID generation

**Comment by Jonah-RF** [comment]
Hey @SamuelHassine, thank you for your feedback! I've pushed new commits to address the issues with ID generation, image naming, and Alpine versions

**Comment by SamuelHassine** [comment]
@Jonah-RF I think we can close this one.

@helene-nguyen can you check the support of proxies in the new version of the RF connector?

**Pull request closed without merging**